src/librustc/session/config.rs

```diff
@@ -109,6 +109,7 @@ impl OutputType {
             OutputType::Bitcode => "bc",
             OutputType::Assembly => "s",
             OutputType::LlvmAssembly => "ll",
+            OutputType::Object if cfg!(target_os = "windows") => "obj",
             OutputType::Object => "o",
```

Comment:
This doesn't feel right, I'd expect to use the target OS not the host one (of the rustc in use).
That is, this won't use `.obj` for cross-compilation and that seems wrong to me.